custom_metrics/javascript.js

```diff
@@ -0,0 +1,81 @@
+return JSON.stringify({
+    'num-ajax-requests': (() => {
+        // Returns the number of ajax requests per page.
+        var ajax_apis = ['xmlhttprequest', 'fetch', 'beacon'];
+        return window.performance.getEntriesByType('resource').filter(r => ajax_apis.includes(r.initiatorType)).length;
+
+    })(),
+    // Returns the number of ajax requests using beacon.
+    'beacon_ajax_usage': window.performance.getEntriesByType('resource').filter(r => r.initiatorType === 'beacon').length,
+    // Returns the number of iframes per page.
+    'iframe': document.getElementsByTagName('iframe').length,
+    'requests_protocol': (() => {
+        // Returns the percentage of http 2 protocol used.
+        try {
+            var requests = $WPT_REQUESTS;
+            if (requests.length > 0) {
+                var ajaxs = window.performance.getEntriesByType('resource').filter(r => ['xmlhttprequest', 'fetch', 'beacon'].includes(r.initiatorType))
+                var resources = window.performance.getEntriesByType('resource').filter(r => r.initiatorType !== 'xmlhttprequest' && r.initiatorType !== 'fetch' && r.initiatorType !== 'beacon')
+                return JSON.stringify({
+                    'total_requests': requests.length,
+                    'total_h1': requests.filter(r => ["http/1.1", "http/1"].includes(r.protocol)).length,
+                    'ajax_h1': ajaxs.filter(r => ["http/1.1", "http/1"].includes(r.nextHopProtocol)).length,
+                    'resources_h1': resources.filter(r => ["http/1.1", "http/1"].includes(r.nextHopProtocol)).length,
+                    'total_h2': requests.filter(r => ["h2", "http/2", "http2"].includes(r.protocol)).length,
+                    'ajax_h2': ajaxs.filter(r => ["h2", "http/2", "http2"].includes(r.nextHopProtocol)).length,
+                    'resources_h2': resources.filter(r => ["h2", "http/2", "http2"].includes(r.nextHopProtocol)).length,
+                    'total_h3': requests.filter(r => r.protocol === "http/3").length,
```

Comment:
There are a few ways H3 can be represented. Would we need to account for those here too?

Context: https://twitter.com/tunetheweb/status/1408026535408381956 cc @tunetheweb 

Are there any other reasons why the sum of each of these metrics wouldn't add up to `total_requests`?

Reply:
added the other representations. Updated the metric here [a6d3ea9](https://github.com/HTTPArchive/legacy.httparchive.org/pull/213/commits/a6d3ea97fb9286033a63d184ebdb2558520a6f9a)